controllers/usersignupcleanup/usersignup_cleanup_controller_test.go

```diff
@@ -251,7 +263,26 @@ func TestUserCleanup(t *testing.T) {
 		require.NoError(t, err)
 		assert.True(t, deleted)
 	})
+}
+
+func alreadyDeletedSignupIgnored(t *testing.T, userSignup *toolchainv1alpha1.UserSignup) {
+	// Now let's simulate the situation when the signup is already being deleted and reconcile it again.
+	nw := corev1.Now()
+	userSignup.DeletionTimestamp = &nw
+	r, req, _ := prepareReconcile(t, userSignup.Name, userSignup)
```

Comment:
btw, it's kind of weird to call `prepareReconcile` here again since it resets the metrics and all the values that were set before in the previous tests, but I understand that for the purpose of the test it shouldn't matter. It's just not clear why the values are 0 when some of them were 1 before that